### PR TITLE
optimize: optimized pattern `.*`, which can be used prefix matching.

### PR DESCRIPTION
**Pull request by membphis**
I do not know if there was any better way, I need a real prefix match.

here are the typical test cases: 

https://github.com/c9s/r3/pull/126/files#diff-54bbbdaaec8855f2ba266946c518709eR15

### CODE REVIEW

File: include/r3.h [hunk 1]
```diff
@@ -202,7 +202,8 @@ int r3_pattern_to_opcode(const char * pattern, unsigned int len);
 
 enum { NODE_COMPARE_STR, NODE_COMPARE_PCRE, NODE_COMPARE_OPCODE };
 
-enum { OP_EXPECT_MORE_DIGITS = 1, OP_EXPECT_MORE_WORDS, OP_EXPECT_NOSLASH, OP_EXPECT_NODASH, OP_EXPECT_MORE_ALPHA };
+enum { OP_EXPECT_MORE_DIGITS = 1, OP_EXPECT_MORE_WORDS, OP_EXPECT_NOSLASH,
+       OP_EXPECT_NODASH, OP_EXPECT_MORE_ALPHA, OP_EXPECT_NOLINEBREAKS};
```
Comment (c9s) [inline include/r3.h]: Can you rename `OP_EXPECT_NOLINEBREAKS` to `OP_GREEDY_ANY` ?

Comment (membphis) [reply include/r3.h]: fixed ^_^

File: tests/check_routes2.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    ck_assert(err == NULL);
+
+    entry = match_entry_create("/foo/bar");
+    matched_route = r3_tree_match_route(n, entry);
+    ck_assert(matched_route != NULL);
+    ck_assert(matched_route->data == &uri0);
+
+    // fixme: should match
+
+    // entry = match_entry_create("/foo");
```
Comment (membphis) [inline tests/check_routes2.c:29]: related PR: https://github.com/c9s/r3/pull/128

File: tests/check_routes2.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#include "config.h"
+#include <stdio.h>
+#include <check.h>
+#include <stdlib.h>
+#include <assert.h>
+#include "r3.h"
+#include "r3_slug.h"
+
+START_TEST (test_routes2s)
```
Comment (c9s) [inline tests/check_routes2.c]: please describe the test case name here, e.g. greedy_pattern

**Review by c9s — APPROVED** [review]

**Pull request merged**